[CI] SearchableSnapshotsIntegTests testCreateAndRestoreSearchableSnapshot failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/6f4r6wd2vwmdc/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests/testCreateAndRestoreSearchableSnapshot

**Reproduction line:**
`./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testCreateAndRestoreSearchableSnapshot" -Dtests.seed=5502EA6EF4776CF7 -Dtests.locale=hu-HU -Dtests.timezone=Etc/GMT+4 -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests&tests.test=testCreateAndRestoreSearchableSnapshot

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <4446 hits>
     but: was null

  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.searchablesnapshots.BaseSearchableSnapshotsIntegTestCase.assertTotalHits(BaseSearchableSnapshotsIntegTestCase.java:302)
  at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testCreateAndRestoreSearchableSnapshot(SearchableSnapshotsIntegTests.java:252)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

The real error (hidden) is the following one:

```
Caused by: org.elasticsearch.search.query.QueryPhaseExecutionException: Query Failed [Failed to execute main query] |  
-- | --
  | at org.elasticsearch.search.query.QueryPhase.executeInternal(QueryPhase.java:205) |  
  | at org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:88) |  
  | at org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:462) |  
  | at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:625) |  
  | at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$2(SearchService.java:487) |  
  | at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:47) |  
  | at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62) |  
  | at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) |  
  | at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33) |  
  | at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:776) |  
  | at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) |  
  | at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) |  
  | at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) |  
  | at java.base/java.lang.Thread.run(Thread.java:833) |  
  | Caused by: java.lang.RuntimeException: Invalid vInt detected (too many bits) |  
  | at org.apache.lucene.store.ByteArrayDataInput.readVInt(ByteArrayDataInput.java:128) |  
  | at org.apache.lucene.codecs.lucene90.Lucene90ScoreSkipReader.readImpacts(Lucene90ScoreSkipReader.java:124) |  
  | at org.apache.lucene.codecs.lucene90.Lucene90ScoreSkipReader$1.getImpacts(Lucene90ScoreSkipReader.java:71) |  
  | at org.apache.lucene.search.MaxScoreCache.getMaxScoreForLevel(MaxScoreCache.java:87) |  
  | at org.apache.lucene.search.MaxScoreCache.getSkipLevel(MaxScoreCache.java:97) |  
  | at org.apache.lucene.search.MaxScoreCache.getSkipUpTo(MaxScoreCache.java:110) |  
  | at org.apache.lucene.search.ImpactsDISI.advanceTarget(ImpactsDISI.java:120) |  
  | at org.apache.lucene.search.ImpactsDISI.advance(ImpactsDISI.java:135) |  
  | at org.apache.lucene.search.ImpactsDISI.nextDoc(ImpactsDISI.java:140) |  
  | at org.apache.lucene.search.Weight$DefaultBulkScorer.scoreRange(Weight.java:276) |  
  | at org.apache.lucene.search.Weight$DefaultBulkScorer.score(Weight.java:254) |  
  | at org.elasticsearch.search.internal.CancellableBulkScorer.score(CancellableBulkScorer.java:45) |  
  | at org.apache.lucene.search.BulkScorer.score(BulkScorer.java:38) |  
  | at org.elasticsearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:194) |  
  | at org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:167) |  
  | at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:555) |  
  | at org.elasticsearch.search.query.QueryPhase.searchWithCollector(QueryPhase.java:232) |  
  | at org.elasticsearch.search.query.QueryPhase.executeInternal(QueryPhase.java:187) |  
  | ... 13 more
```

Looks related to the recent Lucene update

Failed again today: https://gradle-enterprise.elastic.co/s/s5jmbw6ptnawk and it does reproduce locally.
In total, it failed four times in the past three days.

I've run this test locally for 100 times and I got 6 failures out of it, but different from the root one initially reported (`Invalid vInt detected (too many bits)`:

```
REPRODUCE WITH: gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testCreateAndRestoreSearchableSnapshot {seed=[6D26F9A0F2B87774:CF23BDED67F1BE3E]}" -Dtests.seed=6D26F9A0F2B87774 -Dtests.locale=en-SG -Dtests.timezone=Asia/Muscat -Druntime.java=17

org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests > testCreateAndRestoreSearchableSnapshot {seed=[6D26F9A0F2B87774:CF23BDED67F1BE3E]} FAILED
    java.lang.AssertionError:
    Expected: <4613 hits>
         but: was null
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.searchablesnapshots.BaseSearchableSnapshotsIntegTestCase.assertTotalHits(BaseSearchableSnapshotsIntegTestCase.java:302)
        at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testCreateAndRestoreSearchableSnapshot(SearchableSnapshotsIntegTests.java:252)

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=4315, name=Thread-1009, state=RUNNABLE, group=TGRP-SearchableSnapshotsIntegTests]

        Caused by:
        Failed to execute phase [query], all shards failed; shardFailures {[t42d5eTuQnekSjo801Qn7w][rlukvvzzin][0]: org.elasticsearch.transport.RemoteTransportException: [node_s0][127.0.0.1:49255][indices:data/read/search[phase/query]]
        Caused by: org.elasticsearch.search.query.QueryPhaseExecutionException: Query Failed [Failed to execute main query]
                at org.elasticsearch.search.query.QueryPhase.executeInternal(QueryPhase.java:205)
                at org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:88)
                at org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:462)
                at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:625)
                at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$2(SearchService.java:487)
                at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:47)
                at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
                at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
                at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
                at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:776)
                at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
                at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
                at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
                at java.base/java.lang.Thread.run(Thread.java:833)
        Caused by: java.io.EOFException: Reading past end of file [position=14023, length=4385] for clone(_0_Lucene90_0.doc[length=4385, file pointer=4385, offset=616][read seq=no])[length=4385, file pointer=14023, offset=616][read seq=no]
                at org.elasticsearch.xpack.searchablesnapshots.store.input.DirectBlobContainerIndexInput.seekInternal(DirectBlobContainerIndexInput.java:249)
                at org.apache.lucene.store.BufferedIndexInput.seek(BufferedIndexInput.java:318)
                at org.apache.lucene.codecs.lucene90.Lucene90PostingsReader$BlockImpactsDocsEnum.advance(Lucene90PostingsReader.java:1202)
                at org.apache.lucene.search.ImpactsDISI.advance(ImpactsDISI.java:135)
                at org.apache.lucene.search.ImpactsDISI.nextDoc(ImpactsDISI.java:140)
                at org.apache.lucene.search.Weight$DefaultBulkScorer.scoreRange(Weight.java:276)
                at org.apache.lucene.search.Weight$DefaultBulkScorer.score(Weight.java:254)
                at org.elasticsearch.search.internal.CancellableBulkScorer.score(CancellableBulkScorer.java:45)
                at org.apache.lucene.search.BulkScorer.score(BulkScorer.java:38)
                at org.elasticsearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:194)
                at org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:167)
                at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:555)
                at org.elasticsearch.search.query.QueryPhase.searchWithCollector(QueryPhase.java:232)
                at org.elasticsearch.search.query.QueryPhase.executeInternal(QueryPhase.java:187)
                ... 13 more                               
```

--- a/docs/changelog/84341.yaml
+++ b/docs/changelog/84341.yaml
@@ -1,0 +1,6 @@
+pr: 84341
+summary: Fix `DirectBlobContainerIndexInput` cloning method
+area: Snapshot/Restore
+type: bug
+issues:
+ - 84238

--- a/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/input/DirectBlobContainerIndexInput.java
+++ b/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/input/DirectBlobContainerIndexInput.java
@@ -258,19 +258,10 @@ public class DirectBlobContainerIndexInput extends BaseSearchableSnapshotIndexIn
 
     @Override
     public DirectBlobContainerIndexInput clone() {
-        final DirectBlobContainerIndexInput clone = new DirectBlobContainerIndexInput(
-            "clone(" + this + ")",
-            directory,
-            fileInfo,
-            context,
-            stats,
-            position,
-            offset,
-            length,
-            // Clones might not be closed when they are no longer needed, but we must always close streamForSequentialReads. The simple
-            // solution: do not optimize sequential reads on clones.
-            NO_SEQUENTIAL_READ_OPTIMIZATION
-        );
+        final DirectBlobContainerIndexInput clone = (DirectBlobContainerIndexInput) super.clone();
+        // Clones might not be closed when they are no longer needed, but we must always close streamForSequentialReads. The simple
+        // solution: do not optimize sequential reads on clones.
+        clone.sequentialReadSize = NO_SEQUENTIAL_READ_OPTIMIZATION;
         clone.isClone = true;
         return clone;
     }

